rust/skim/src/lib.rs

```diff
@@ -10,27 +10,38 @@ mod ffi {
 }
 
 struct Item {
-    text: String,
+    text_no_newlines: String,
+    text_lower: String,
     orig_text: String,
 }
 impl Item {
     fn new(text: String) -> Self {
         return Self{
             // Text that will be shown should not contains new lines since in this case skim may
             // live some symbols on the screen, and this looks odd.
-            text: text.replace("\n", " "),
+            text_no_newlines: text.replace("\n", " "),
+            // This will be used to match with, since CaseMatching::Ignore does not the right
+            // thing, so the text is converted to lower register, so as the query itself.
+            text_lower: text.to_lowercase(),
```

Comment:
we want to transform `text_no_newlines` to lower, no?

Reply:
This is what will be rendered by the skim, so this is what you will see when you will use fuzzy search, and it should had the same case as the original query. So the answer is no.

Reply:
I meant that `text_lower: text_no_newlines.to_lowercase()` but maybe I misunderstood how this works.

Reply:
Oh, I see. This is not important actually, since the only difference between what is written and you suggestion is what will skim use for matching, query with new lines or not, and this does not makes any difference for fuzzy searching since you cannot search by newline anyway.

Just to clarify, here is an example, consider the following entry in the history:

### `orig_text`

This is used for selection when the entry had been selected by user.

```sql
CREATE TABLE foo
(
  key Int,
  value Int
)
Engine=Null()
```

### `text_no_newlines`

This is used for rendering in skim, new lines left some symbols on the screen otherwise

```sql
CREATE TABLE foo (  key Int,  value Int ) Engine=Null()
```

### `text_lower`

And this, is what will be used to matching with skim, and here new lines or spaces does not makes any difference.

```sql
create table foo
(
  key int,
  value int
)
engine=null()
```

Reply:
Hope this clarify things.

Reply:
> since you cannot search by newline anyway

This detail was the important one which I wasn't sure about, thanks!